NICE.Identity.Management/ClientApp/src/components/App/App.tsx

```diff
@@ -18,21 +18,61 @@ import { SelectRoles } from "./../../views/SelectRoles/SelectRoles";
 import { EditUser } from "./../../views/EditUser/EditUser";
 import { DeleteOrganisation } from "./../../views/DeleteOrganisation/DeleteOrganisation";
 
+import { Endpoints } from "../../data/endpoints";
+import { fetchData } from "../../helpers/fetchData";
+import { isDataError } from "../../helpers/isDataError";
+
+export type MyAccountDetails = {
+	displayName: string,
+	links: []
+};
+
 export class App extends React.Component {
-	render(): JSX.Element {
+	state = {
+		isLoading: false,
+		auth: {} as IdamProviderProps
+	}
+	async componentDidMount(): Promise<void> {
+		const fetchOptions = {
+			method: "GET",
+			headers: { "Content-Type": "application/json" }
+		};
+		const myAccountDetails = await fetchData(
+			Endpoints.identityManagementUser,
+			fetchOptions,
+		) as MyAccountDetails;
+
 		const auth: IdamProviderProps = {
 			links: [
+				{ text: myAccountDetails.displayName, url: "#" },
 				{ text: "Health checks", url: "/healthchecks-ui" },
 				{ text: "Sign out", url: "/Account/Logout" },
 			],
-			displayName: "John",
+			displayName:  myAccountDetails.displayName,
 			provider: "idam",
 		};
 
+		this.setState({
+			isLoading: true,
+			auth: auth
+		});
+
+		if (isDataError(myAccountDetails)) {
+			this.setState({ error: myAccountDetails });
+		}
+
+		this.setState({
+			isLoading: false
+		});
```

Comment:
```suggestion
		this.setState({
			auth: auth,
			isLoading: false
		});
```

Move the setState auth from above down here.